R/Arena.R

```diff
@@ -0,0 +1,49 @@
+#' R6 Class representing an arena
+#'
+#' An arena has a representation in text.
+Arena <- R6::R6Class("Arena", #nolint
+  private = list(
+    .text = NULL,
+    check = function(text) {
+      if (!is.character(text)) {
+        stop("text must be of character type")
+      }
+      if (length(text) < 3) {
+        stop("arena must have at least 3 rows")
+      }
+      if (any(nchar(gsub("X|\\.", "", text)) != 0)) {
+        stop("text must contain only `X` and `.`")
+      }
+      if (any(nchar(text) < 3)) {
+        stop("Each row in the arena must have at least 3 characters")
+      }
+      if (length(unique(nchar(text))) != 1) {
+        stop("Each row in the arena must have the same number of characters")
+      }
+    }
+  ),
+  active = list(
```

Comment:
I would be interested to hear what the `active` part of an R6 class is. Now I will assume it is brilliant :-)